### PR TITLE
COMMERCE-9637 Refactoring COMMERCE-6656 and COMMERCE-6657 tests

### PR DESCRIPTION
**Pull request by kristinonias**
[COMMERCE-9637](https://issues.liferay.com/browse/COMMERCE-9637)
[COMMERCE-9638](https://issues.liferay.com/browse/COMMERCE-9638)

### CODE REVIEW

**Comment by nikki-pru** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductCard.testcase [hunk 1]
```diff
@@ -1064,160 +1064,210 @@ definition {
...
+			Type(
+				key_fieldLabel = "Base Price",
+				locator1 = "TextInput#GENERIC_TEXT_INPUT",
+				value1 = "10.0");
 
-		Button.click(button = "Publish");
+			Button.click(button = "Publish");
 
-		WaitForElementNotPresent(locator1 = "Message#SUCCESS");
+			WaitForElementNotPresent(locator1 = "Message#SUCCESS");
```
Comment (nikki-pru) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductCard.testcase:1214]: Also looks like a single goal, might be better off as a macro

**Review by nikki-pru — CHANGES_REQUESTED** [review]
@kristinonias , please see lines that could be changed into macros instead of floating functions in a test.

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductCard.testcase [hunk 2]
```diff
@@ -1064,160 +1064,210 @@ definition {
...
+			Check.checkToggleSwitch(
+				key_toggleSwitchLabel = "Required",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
 
-		Check.checkToggleSwitch(
-			key_toggleSwitchLabel = "SKU Contributor",
-			locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
+			Check.checkToggleSwitch(
+				key_toggleSwitchLabel = "SKU Contributor",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
```
[the remark this answers is not in the document]
Comment (kristinonias) [reply modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductCard.testcase]: Fixed!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductCard.testcase [hunk 3]
```diff
@@ -1064,160 +1064,210 @@ definition {
...
+				optionName = "Color");
+		}
 
-		Check.checkToggleSwitch(
-			key_toggleSwitchLabel = "Use in Faceted Navigation",
-			locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
+		task ("And Faceted Navigation, Required, and SKU Contributor are enabled with an option value") {
+			Check.checkToggleSwitch(
+				key_toggleSwitchLabel = "Use in Faceted Navigation",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
```
[the remark this answers is not in the document]
Comment (kristinonias) [reply modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductCard.testcase]: Fixed!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductCard.testcase [hunk 4]
```diff
@@ -1064,160 +1064,210 @@ definition {
...
 
-		CommerceEntry.addOptionValue(
-			valueName = "Black",
-			valuePosition = "");
+			CommerceEntry.addOptionValue(
+				valueName = "Black",
+				valuePosition = "");
 
-		Button.click(button = "Submit");
+			Button.click(button = "Submit");
```
[the remark this answers is not in the document]
Comment (kristinonias) [reply modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductCard.testcase]: Fixed!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductCard.testcase [hunk 5]
```diff
@@ -1064,160 +1064,210 @@ definition {
...
+			Select(
+				key_fieldLabel = "Price Type",
+				locator1 = "Select#GENERIC_SELECT_FIELD",
+				value1 = "Static");
 
-		Button.clickSave();
+			Button.clickSave();
 
-		Alert.viewSuccessMessage();
+			Alert.viewSuccessMessage();
```
[the remark this answers is not in the document]
Comment (kristinonias) [reply modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductCard.testcase]: Created CommerceOptions.setPriceType.

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductCard.testcase [hunk 6]
```diff
@@ -1064,160 +1064,210 @@ definition {
...
+			Type(
+				key_fieldLabel = "SKU",
+				locator1 = "TextInput#GENERIC_TEXT_INPUT",
+				value1 = "MIN55861");
 
-		Button.clickSave();
+			Button.clickSave();
 
-		Alert.viewSuccessMessage();
+			Alert.viewSuccessMessage();
```
[the remark this answers is not in the document]
Comment (kristinonias) [reply modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductCard.testcase]: Created CommerceOptions.setDeltaPrice macro.

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductCard.testcase (hunk 1 above)
Comment (kristinonias) [reply modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductCard.testcase]: Created CommerceOptions.enableUseInFacetedNavigation, CommerceOptions.enableRequired, and CommerceOptions.enableSKUContributor macros.

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductCard.testcase [hunk 7]
```diff
@@ -1064,160 +1064,210 @@ definition {
...
-			key_title = "Add Options",
-			locator1 = "CommerceEntry#ADD_RELATION_FIELD",
-			value1 = "Color");
+			Type(
+				key_title = "Add Options",
+				locator1 = "CommerceEntry#ADD_RELATION_FIELD",
+				value1 = "Color");
 
-		Button.click(button = "Select");
+			Button.click(button = "Select");
```
[the remark this answers is not in the document]
Comment (kristinonias) [reply modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductCard.testcase]: Fixed!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductCard.testcase [hunk 8]
```diff
@@ -1064,160 +1064,210 @@ definition {
...
+			Check.checkToggleSwitch(
+				key_toggleSwitchLabel = "Use in Faceted Navigation",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
 
-		Check.checkToggleSwitch(
-			key_toggleSwitchLabel = "Required",
-			locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
+			Check.checkToggleSwitch(
+				key_toggleSwitchLabel = "Required",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
```
[the remark this answers is not in the document]
Comment (kristinonias) [reply modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductCard.testcase]: Fixed!

**Comment by nikki-pru** [comment]
ci:test:commerce-functional

**Comment by nikki-pru** [comment]
**PASSED**:
[LocalFile.CommerceProductCard#AssertDynamicPriceBundledProductCannotBeAddedToCartDirectly](https://testray.liferay.com/home/-/testray/case_results/1915955348)
[LocalFile.CommerceProductCard#AssertStaticPriceBundledProductCannotBeAddedToCartDirectly](https://testray.liferay.com/home/-/testray/case_results/1915955424)

**Comment by nikki-pru** [comment]
ci:forward

**Pull request closed without merging**